undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java

```diff
@@ -982,6 +983,10 @@ private DeploymentInfo createServletConfig() throws StartException {
                 webSocketDeploymentInfo.setBuffers(servletContainer.getWebsocketsBufferPool().getValue());
                 webSocketDeploymentInfo.setWorker(servletContainer.getWebsocketsWorker().getValue());
                 webSocketDeploymentInfo.setDispatchToWorkerThread(servletContainer.isDispatchWebsocketInvocationToWorker());
+
+                // Enables compression. This fixes WFLY-6787
+                webSocketDeploymentInfo.addExtension(new PerMessageDeflateHandshake(false));
```

Comment:
Thinking about it I am ok with this going in. If it causes problems for anyone they can use a configurator to work around them programmatically, so it should be fine.